icu4c/source/common/uloc.cpp

```diff
@@ -1738,7 +1738,7 @@ _canonicalize(const char* localeID,
         len = (int32_t)uprv_strlen(d);
 
         if (name != NULL) {
-            uprv_strncpy(name, d, len);
+            uprv_memcpy(name, d, len);
```

Comment:
How about,

```cpp
uprv_strncpy(name, d, nameCapacity);
```

and remove line 1738?

Reply:
We still need "len" to be updated for the rest of the code. So is it Ok to leave it like this ?